core/gallery/backends.go

```diff
@@ -383,3 +394,180 @@ func RegisterBackends(systemState *system.SystemState, modelLoader *model.ModelL
 
 	return nil
 }
+
+// ResolveBestBackendName returns the concrete backend name to use for a given meta or concrete backend name,
+// based on the current system state and gallery metadata. If the provided name is already concrete, it is returned as-is.
+func ResolveBestBackendName(galleries []config.Gallery, systemState *system.SystemState, name string) (string, error) {
+	backends, err := AvailableBackends(galleries, systemState)
+	if err != nil {
+		return "", err
+	}
+	be := FindGalleryElement(backends, name)
+	if be == nil {
+		return "", fmt.Errorf("no backend found with name %q", name)
+	}
+	if !be.IsMeta() {
+		return be.Name, nil
+	}
+	best := be.FindBestBackendFromMeta(systemState, backends)
+	if best == nil {
+		return "", fmt.Errorf("no backend found with capabilities %v", be.CapabilitiesMap)
+	}
+	return best.Name, nil
+}
+
+// ListSystemBackendsSelected lists system backends and, when multiple concrete backends share the same alias
+// (e.g., cpu-llama-cpp and cuda12-llama-cpp both alias to "llama-cpp"), selects the optimal one based on the
+// detected system capability (GPU vendor/platform). Concrete backend names are always included.
+func ListSystemBackendsSelected(systemState *system.SystemState) (SystemBackends, error) {
```

Comment:
at this point would make sense to actually modify directly `ListSystemBackends` 

https://github.com/mudler/LocalAI/blob/21faa4114bf6c8980fc612e7db5a2a13b62e8d23/core/gallery/backends.go#L284

Its usage in the code is quite limited https://github.com/search?q=repo%3Amudler%2FLocalAI%20ListSystemBackends&type=code

otherwise would make sense to re-use it as much as possible, to avoid code dups